In [4]:
import os
os.chdir("/Users/srujansingh/Documents/JeanFan_Lab/Kidney_Structural_Analysis/")

In [7]:
import scanpy as sc
ctrl1_data = sc.read_h5ad("data/Ctrl_1A2_visiumhd_2um_proximal_labeled.h5ad")

AnnData object is a standardized data structure used in SC and ST datasets that hold all information at one pleace like:

Observations (obs) → cells / spots / pixels

Variables (var) → genes

Matrix (X) → expression values (genes × cells/spots)

Extra metadata → spatial coordinates, annotations, embeddings, etc.


In [8]:
ctrl1_data

AnnData object with n_obs × n_vars = 5549563 × 19059
    obs: 'in_tissue', 'array_row', 'array_col', 'pxl_row_in_fullres', 'pxl_col_in_fullres', 'proximal_tubule_geojson_feature_index'
    var: 'gene_ids', 'feature_types', 'genome'
    obsm: 'spatial'

The ctr1_data AnnData object contaisn 19059 genes (features) x 5549563 bins (spatial units).

In [9]:
ctrl1_data.obs.head()

,in_tissue,array_row,array_col,pxl_row_in_fullres,pxl_col_in_fullres,proximal_tubule_geojson_feature_index
s_002um_00000_02609-1,1,0,2609,1164.609932,6249.759373,<NA>
s_002um_00000_02610-1,1,0,2610,1164.662587,6245.220638,<NA>
s_002um_00000_02611-1,1,0,2611,1164.715242,6240.681902,<NA>
s_002um_00000_02612-1,1,0,2612,1164.767897,6236.143167,<NA>
s_002um_00000_02613-1,1,0,2613,1164.820552,6231.604433,<NA>


In [11]:
ctrl1_data.obs["proximal_tubule_geojson_feature_index"].unique()

<IntegerArray>
[<NA>,  929,  145, 2875, 5680, 2409, 2683,   86, 4813,  509,
 ...
 2607, 1603, 4799, 1242, 2392, 3531, 6605, 5953, 1777, 5938]
Length: 1598, dtype: Int64